### ISSUE TITLE
New sanitizer: delete-tags

### ISSUE THREAD

**Issue by lonvia** [osm-search/Nominatim#2953]
In order to filter out odd tagging practises like the postbox refs in Germany
(see #2652) we could do with a sanitizer that removes some names and address parts from
the list of indexed names. It should work along those lines:

### Sanitizer 'delete-tags'

Removes a name or address name from the list, when it matches all properties
given in the parameter list. There are two kinds of properties. Some refer to
the name and some refer to the place being indexed. If you specify only
place properties, then all names and address parts will be deleted and the
place will effectively not be searchable.

Where parameters contain regular expressions, they are always matched against
the full string. Add an exclamation mark (!) in front of the expression to
negate the match, i.e. delete the tag only if the property does not match.

When a parameter contains a list of conditions, a single match is sufficient.

Parameters:

```
 type - Either 'name' or 'address'.
 kind - A single string or a list of strings containing regular
        expressions that are matched against the 'kind' property of the name.
 suffix - A single string or a list of strings containing regular
          expressions that are matched against the 'suffix' property.
 name - A single string or a list of strings containing regular
        expressions that are matched against 'name' property of the name.
 country_code - A single string or list of strings containing two-letter
                lower-case country codes. Only places within one of the listed
                countries will be deleted.
 rank_address - A single string or list of strings containing either a single
                number referring to a rank or a range of the form <from>-<to>.
```

**Comment by lonvia** [comment]
If somebody wants to take this up, here are some hints to get you started:

* there is a tutorial on creating sanitizers at https://nominatim.org/tutorials/customize-transliteration.html
* the reference for sanitizers can be found at https://nominatim.org/release-docs/latest/develop/ICU-Tokenizer-Modules/#custom-sanitizer-modules
*  have a look at the existing sanitizers in https://github.com/osm-search/Nominatim/tree/master/nominatim/tokenizer/sanitizers. This is also where the new sanitizer should go.
* unit tests for existing sanitizers are in https://github.com/osm-search/Nominatim/tree/master/test/python/tokenizer/sanitizers. Add a new test for the new sanitizer.

**Comment by SainiAditya1** [comment]
assign this to me , I would like to work on this ..

**Comment by lonvia** [comment]
We do not assign tasks. Just pick what interests you. Please make sure to read the [GSOC instructions](https://wiki.openstreetmap.org/wiki/User:Lonvia/GSoC_2021_Nominatim_Projects).

**Comment by biswajit-k** [comment]
Hi @lonvia, I was going through the above issue, couldn't clearly get what `type` means, is it refering to the `names` and `address` lists on the `ProcessInfo` object or the `type` field on the `PlaceInfo` object(if so, then what does name and address mean there)?

**Comment by lonvia** [comment]
The `type` indeed is meant to refer to the `names` and `address` lists in the `ProcessInfo` object. Depending on the `type` the function should clean out one or the other list.

**Comment by biswajit-k** [comment]
Thanks for clarifying @lonvia, I have added a PR for the same. I request you to please have a look.

**Comment by lonvia** [comment]
Implemented in #2993.
